### PR TITLE
Bundle with tsup and replace babel-jest

### PR DESCRIPTION
**Pull request by ndelangen**
Original PR: https://github.com/storybookjs/test-runner/pull/224 by @bryanjtc 👏 

Bundles code and some dependencies with tsup.
Replaces babel-jest with @swc/core. With swc tests should run faster than using babel.
Adds require.resolve for jest-junit and jest-serializer-html.
Adds support for esm only projects that don't use a custom config file.
Tested locally using test-storybook:ci commands and some projects that I have with storybook and test-runner.

<details>
  <summary>📦 Published PR as canary version: <code>0.10.1--canary.250.6355af6.0</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @storybook/test-runner@0.10.1--canary.250.6355af6.0
  # or 
  yarn add @storybook/test-runner@0.10.1--canary.250.6355af6.0
  ```
</details>

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -124,7 +122,9 @@
     "jest-watch-typeahead": "^2.0.0",
     "node-fetch": "^2",
     "playwright": "^1.14.0",
-    "regenerator-runtime": "^0.13.9"
+    "regenerator-runtime": "^0.13.9",
```
Comment (ndelangen) [inline package.json:125]: Can we remove regenerator-runtime?

**Comment by bryanjtc** [comment]
Update on this?

**Comment by bryanjtc** [comment]
@yannbf What are the next steps for this pr now that storybook v7 is released? I can separate the pr for better testing.

**Comment by ndelangen** [comment]
@yannbf let's look at this together, and merge/release it today if we can?

**Comment by yannbf** [comment]
Hey there! Thanks a lot for your patience, I truly appreciate it.

Sorry I'm on vacation and will be back next Tuesday. Now that v7 is released, we can get this ready to merge into `next` so we can try it out in a new release. If you want to go to town with it @ndelangen feel free (but make sure test this well), else we can do it when I'm back

A couple of things:
1. For some reason, the `@storybook/preview-api` dependency got removed and `@storybook/store` got added. That seems to be incorrect
2. Would it be possible to use the `tsup-node` binary instead of `tsup`? The end result in `dist` is 10000% more friendly when not prebundling the dependencies, like a 600 loc vs 48000 loc.  Do we really need to prebundle the dependencies?
3. Are we sure about the implications of babel-jest with swc? Is it a 100% compatible change? Else this means a breaking change

**Comment by bryanjtc** [comment]
I can move the dev dependencies that are being prebundled to dependencies. Tsup doesn't bundle dependencies. Replacing babel-jest with swc shouldn't be a breaking change. Swc jest is designed to be a drop-in replacement.

**Comment by bryanjtc** [comment]
@yannbf Fixed everything. I also disabled code splitting, minification, treeshake and bundle in the tsup config. It should be easier to read and modify.

**Comment by yannbf** [comment]
This is looking amazing, thank you once again @bryanjtc !! <3 <3

**Review by yannbf — APPROVED** [review]

**Pull request merged**